### ISSUE TITLE
Merge API documentation and delete old versions 

### ISSUE THREAD

**Issue by muneeb-ali** [blockstack/blockstack-core#510]
We have multiple versions of our API documentation right now (some are out sync with each other). Let's 

- [ ] Pick one place for API documentation 
- [ ] Delete old places/branches 
- [ ] Make the API documentation easier to discover

**Comment by vbrandl** [comment]
I started writing a client library to use the Blockstack API and I came across the following issues until know:

- The return values aren't documented. There are examples for all return values but no formal documentation that tells the datatype or which attributes are optional or required. It would be best to have the return structures documented like this:

Attribute | Datatype | Required
--- | --- | ---
passwd | String | Y
port | u16 | Y
regtest | bool | N
server | String | Y

- Inconsistent data types. Sometimes `true` is passed as a string `"True"` and sometimes it is passed as a boolean value. Same problem with number types. `"1234"` vs `1234`. There should be a consistent pattern. Either everything gets passed as a string or primitive types _never_ get passed as strings.

That's it for now. I'll post again if I stumble across more problems.
